pkg/pillar/cmd/zedrouter/metadata_handlers.go

```diff
@@ -0,0 +1,693 @@
+	hdl.zedrouter.log.Tracef("wwanAppInfoHandler.ServeHTTP")
+	w.Header().Add("Content-Type", "application/json")
+
+	remoteIP := net.ParseIP(strings.Split(r.RemoteAddr, ":")[0])
+	anStatus := hdl.zedrouter.lookupAppNetworkStatusByAppIP(remoteIP)
+	if anStatus == nil {
+		hdl.zedrouter.log.Errorf("Could not find network instance by ip %v", remoteIP)
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("not found"))
+		return
+	}
+
+	diskStatusList := hdl.zedrouter.lookupDiskStatusList(
+		anStatus.UUIDandVersion.UUID.String())
+
+	var appInfo types.AppInfo
+	for _, st := range diskStatusList {
+		if st.Devtype != "AppCustom" {
+			continue
+		}
+
+		blob := types.AppBlobsAvailable{
+			CustomMeta: st.CustomMeta,
+			DownloadURL: fmt.Sprintf("http://169.254.169.254/eve/app-custom-blobs/%s",
+				st.DisplayName),
+		}
+
+		appInfo.AppBlobs = append(appInfo.AppBlobs, blob)
+	}
+
+	resp, _ := json.Marshal(appInfo)
+	w.WriteHeader(http.StatusOK)
+	w.Write(resp)
+}
+
+func (hdl AppCustomBlobsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	blobName := path.Base(r.URL.Path)
+
+	remoteIP := net.ParseIP(strings.Split(r.RemoteAddr, ":")[0])
+	anStatus := hdl.zedrouter.lookupAppNetworkStatusByAppIP(remoteIP)
+	if anStatus == nil {
+		hdl.zedrouter.log.Errorf("Could not find network instance by ip %v", remoteIP)
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("not found"))
+		return
+	}
+
+	diskStatusList := hdl.zedrouter.lookupDiskStatusList(
+		anStatus.UUIDandVersion.UUID.String())
+
+	var blobFileLocation string
+	for _, st := range diskStatusList {
+		if st.Devtype != "AppCustom" {
+			continue
+		}
+
+		if st.DisplayName == blobName {
+			blobFileLocation = st.FileLocation
+			break
+		}
+	}
+
+	if blobFileLocation == "" {
+		http.Error(w, r.RequestURI, http.StatusNotFound)
+		return
+	}
+
+	f, err := os.Open(blobFileLocation)
+	if err != nil {
+		http.Error(w, r.RequestURI, http.StatusNotFound)
+		return
+	}
+	defer f.Close()
+	fi, err := f.Stat()
+	if err != nil {
+		http.Error(w, r.RequestURI, http.StatusNotFound)
+		return
+	}
+	modTime := fi.ModTime()
+
+	http.ServeContent(w, r, blobFileLocation, modTime, f)
+}
+
+func HandlePatchDescription(z *zedrouter) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// WithPatchEnvelopesByIP middleware returns envelopes which are more than 0
+		envelopes := r.Context().Value("patchEnvelopes").([]types.PatchEnvelopeInfo)
+
+		b, err := json.Marshal(envelopes)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusUnprocessableEntity)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(b)
+		return
+	}
+}
+
+func sendError(w http.ResponseWriter, code int, msg string) {
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(code)
+	w.Write([]byte(fmt.Sprintf("{\"message\": \"%s\"}", msg)))
+}
+
+func HandlePatchDownload(z *zedrouter) func(http.ResponseWriter, *http.Request) {
```

Comment:
Still some revive and codespell issues from yetus to fix.

Reply:
Updated

Reply:
But still more below.

Reply:
:smiling_face_with_tear:

Reply:
Let's see what yetus have to say now